src/main/java/org/helpberkeley/memberdata/WorkflowParser.java

```diff
@@ -51,23 +51,18 @@ protected WorkflowParser(final String csvData) {
     public static WorkflowParser create(Map<String, Restaurant> globalRestaurants, String csvData) {
 
         ControlBlock controlBlock = ControlBlock.create(csvData);
+        String version = controlBlock.getVersion();
         WorkflowParser workflowParser;
 
-        switch (controlBlock.getVersion()) {
-            case Constants.CONTROL_BLOCK_VERSION_UNKNOWN:
-                throw new MemberDataException("Control block not found");
-            case Constants.CONTROL_BLOCK_VERSION_200:
-                workflowParser = new WorkflowParserV200(csvData);
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_300:
-            case Constants.CONTROL_BLOCK_VERSION_301:
-            case Constants.CONTROL_BLOCK_VERSION_302:
-                workflowParser = new WorkflowParserV300(csvData);
-                break;
-            case Constants.CONTROL_BLOCK_VERSION_1:
-            default:
-                throw new MemberDataException(
-                        "Control block version " + controlBlock.getVersion() + " is not supported.\n");
+        if (version.equals(Constants.CONTROL_BLOCK_VERSION_UNKNOWN)) {
+            throw new MemberDataException("Control block not found");
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_200)) {
+            workflowParser = new WorkflowParserV200(csvData);
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_300)) {
+            workflowParser = new WorkflowParserV300(csvData);
+        } else {
+            throw new MemberDataException(MessageFormat.format(
+                    ControlBlock.UNSUPPORTED_VERSION_GENERIC, version));
```

Comment:
Code coverage says that this throw is not being tested.  Is this one of the cases where we can't currently get here?

Reply:
yeah doesn't look like it's possible. the only ways to get here are from DriverPostFormatV200/V300 which have already been checked or from oneKitchenDriverHistory, which I don't think takes any input or could ever return a v1 control block